pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -3860,106 +3860,2824 @@ func Test_server_StreamInsert(t *testing.T) {
+	generator, _ := vector.Uint8VectorGenerator(dist)
+	ivecs := generator(num, dim)
+	result := make([][]float32, num)
+
+	for j, ivec := range ivecs {
+		vec := make([]float32, dim)
+		for i := 0; i < dim; i++ {
+			vec[i] = float32(ivec[i])
+		}
+		result[j] = vec
+	}
+	return result
+}
+
 func Test_server_MultiInsert(t *testing.T) {
 	t.Parallel()
+
 	type args struct {
 		ctx  context.Context
 		reqs *payload.Insert_MultiRequest
 	}
 	type fields struct {
 		name              string
 		ip                string
-		ngt               service.NGT
-		eg                errgroup.Group
 		streamConcurrency int
+		svcCfg            *config.NGT
+		svcOpts           []service.Option
 	}
 	type want struct {
-		wantRes *payload.Object_Locations
-		err     error
+		wantRes    *payload.Object_Locations
+		err        error
+		containErr []error // check the function output error contain one of the error or not
 	}
 	type test struct {
 		name       string
 		args       args
 		fields     fields
 		want       want
 		checkFunc  func(want, *payload.Object_Locations, error) error
-		beforeFunc func(args)
+		beforeFunc func(*testing.T, *server)
 		afterFunc  func(args)
 	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	type objectType int
+	const (
+		Uint8 objectType = iota
+		Float
+	)
+
+	// common variables for test
+	const (
+		name      = "vald-agent-ngt-1" // agent name
+		id        = "uuid-1"           // insert request id
+		intVecDim = 3                  // int vector dimension
+		f32VecDim = 3                  // float32 vector dimension
+		maxVecDim = 1 << 18            // reference value for testing, this value is temporary
+	)
+	var (
+		ip = net.LoadLocalIP() // agent ip address
+
+		// default NGT configuration for test
+		defaultIntSvcCfg = &config.NGT{
+			Dimension:    intVecDim,
+			DistanceType: ngt.Angle.String(),
+			ObjectType:   ngt.Uint8.String(),
+			KVSDB:        &config.KVSDB{},
+			VQueue:       &config.VQueue{},
+		}
+		defaultF32SvcCfg = &config.NGT{
+			Dimension:    f32VecDim,
+			DistanceType: ngt.Angle.String(),
+			ObjectType:   ngt.Float.String(),
+			KVSDB:        &config.KVSDB{},
+			VQueue:       &config.VQueue{},
+		}
+		defaultSvcOpts = []service.Option{
+			service.WithEnableInMemoryMode(true),
+		}
+	)
+
+	genMultiInsertReq := func(t objectType, dist vector.Distribution, num int, dim int, cfg *payload.Insert_Config) *payload.Insert_MultiRequest {
+		var vecs [][]float32
+		switch t {
+		case Float:
+			vecs = genF32Vec(dist, num, dim)
+		case Uint8:
+			vecs = genIntVec(dist, num, dim)
+		}
+
+		req := &payload.Insert_MultiRequest{
+			Requests: make([]*payload.Insert_Request, num),
+		}
+		for i, vec := range vecs {
+			req.Requests[i] = &payload.Insert_Request{
+				Vector: &payload.Object_Vector{
+					Id:     "uuid-" + strconv.Itoa(i+1),
+					Vector: vec,
+				},
+				Config: cfg,
+			}
+		}
+
+		return req
+	}
```

Comment:
why I define it in a function is because it is the function to generate MultiInsert request which can be only used in this function. Do you think I still need to define it as a global function?

Reply:
fixed :)